### PR TITLE
implement list role assignments & list project role assignments

### LINKED ISSUE #1257: Add capability to list/search principal <--> role assignments

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1257 

As usual, I've broken my PR into discrete commits for API changes and corresponding SDK changes and CLI changes. The easiest way to review might be to review each of those as a unit.

~~Built on top of #1369, which should be merged first.~~ Done.

### CODE REVIEW

File: sdk/v2/authz/role_assignments.go [hunk 1]
```diff
@@ -2,27 +2,78 @@ package authz
...
+	)
+}
+
+// RoleAssignmentsSelector represents useful filter criteria when selecting
+// multiple RoleAssignments for API group operations like list.
+type RoleAssignmentsSelector struct {
+	// Principal specifies that only RoleAssignments for the specified Principal
+	// should be selected.
+	Principal *libAuthz.PrincipalReference
+	// TODO: Document this
```
Comment (vdice) [inline sdk/v2/authz/role_assignments.go:58]: Could probably address this now?

**Review by vdice — APPROVED** [review]
LGTM!

File: sdk/v2/authz/role_assignments.go (hunk 1 above)
Comment (krancour) [reply sdk/v2/authz/role_assignments.go:58]: Whoops. Good eye!

**Pull request merged**